lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -532,133 +550,165 @@ class PullbackInfo {
-      pullbackStruct->setGenericParams(genericParams);
-      pullbackStruct->setGenericEnvironment(
-          vjpGenericSig->createGenericEnvironment());
+          cloneGenericParameters(astCtx, linearMapStruct, genericSig);
+      linearMapStruct->setGenericParams(genericParams);
+      linearMapStruct->setGenericEnvironment(
+          genericSig->createGenericEnvironment());
     }
-    pullbackStruct->setBraces(loc);
+    linearMapStruct->setBraces(loc);
     computeAccessLevel(
-        pullbackStruct, original->getEffectiveSymbolLinkage());
-    pullbackStruct->computeType();
-    assert(pullbackStruct->hasInterfaceType());
-    file.addVisibleDecl(pullbackStruct);
+        linearMapStruct, original->getEffectiveSymbolLinkage());
+    linearMapStruct->computeType();
+    assert(linearMapStruct->hasInterfaceType());
+    file.addVisibleDecl(linearMapStruct);
     LLVM_DEBUG({
       auto &s = getADDebugStream();
-      s << "Pullback struct created for function @" << original->getName()
+      std::string structName;
+      switch (kind) {
+      case AutoDiffAssociatedFunctionKind::JVP:
+        structName = "Differential";
+        break;
+      case AutoDiffAssociatedFunctionKind::VJP:
+        structName = "Pullback";
+        break;
+      }
+      s << structName << " struct created for function @" << original->getName()
         << " bb" << originalBB->getDebugID() << '\n';
-      pullbackStruct->print(s);
+      linearMapStruct->print(s);
       s << '\n';
     });
-    return pullbackStruct;
+    return linearMapStruct;
   }
 
 public:
-  PullbackInfo(const PullbackInfo &) = delete;
-  PullbackInfo &operator=(const PullbackInfo &) = delete;
+  LinearMapInfo(const LinearMapInfo &) = delete;
+  LinearMapInfo &operator=(const LinearMapInfo &) = delete;
 
-  explicit PullbackInfo(ADContext &context, SILFunction *original,
-                        SILFunction *vjp, const SILAutoDiffIndices &indices);
+  explicit LinearMapInfo(
+      ADContext &context, AutoDiffAssociatedFunctionKind kind,
+      SILFunction *original, SILFunction *assocFn,
+      const SILAutoDiffIndices &indices);
 
-  /// Returns the pullback struct associated with the given original block.
-  StructDecl *getPullbackStruct(SILBasicBlock *origBB) const {
-    return pullbackStructs.lookup(origBB);
+  /// Returns the linear map struct associated with the given original block.
+  StructDecl *getLinearMapStruct(SILBasicBlock *origBB) const {
+    return linearMapStructs.lookup(origBB);
   }
 
-  /// Returns the lowered SIL type of the pullback struct associated with the
+  /// Returns the lowered SIL type of the linear map struct associated with the
   /// given original block.
-  SILType getPullbackStructLoweredType(SILBasicBlock *origBB) const {
-    auto *pbStruct = getPullbackStruct(origBB);
+  SILType getLinearMapStructLoweredType(SILBasicBlock *origBB) const {
+    auto *pbStruct = getLinearMapStruct(origBB);
```

Comment:
```suggestion
    auto *linMapStruct = getLinearMapStruct(origBB);
```

Please generalize all references to "pullback"/"pb" in `LinearMapInfo` to "linearMap"/"linMap".